storage_drivers/fake/fake.go

```diff
@@ -561,6 +567,51 @@ func (d *StorageDriver) Publish(name string, publishInfo *utils.VolumePublishInf
 	return errors.New("fake driver does not support Publish")
 }
 
+// CreateSnapshot creates a snapshot for the given volume
+func (d *StorageDriver) CreateSnapshot(snapshotName string, volConfig *storage.VolumeConfig) (
+	*storage.Snapshot, error) {
+
+	internalVolName := volConfig.InternalName
+
+	if d.Config.DebugTraceFlags["method"] {
+		fields := log.Fields{
+			"Method":       "CreateSnapshot",
+			"Type":         "StorageDriver",
+			"snapshotName": snapshotName,
+			"sourceVolume": internalVolName,
+		}
+		log.WithFields(fields).Debug(">>>> CreateSnapshot")
+		defer log.WithFields(fields).Debug("<<<< CreateSnapshot")
+	}
+
+	// Ensure source volume exists
+	vol, ok := d.Volumes[internalVolName]
+	if !ok {
+		return nil, fmt.Errorf("source volume %s not found", internalVolName)
+	}
+	// Initialize snapshots list for a snapshot with same name
+	if _, ok := d.Snapshots[snapshotName]; ok {
+		return nil, fmt.Errorf("snapshot %s already exists", snapshotName)
+	}
+
+	snapshot := &storage.Snapshot{
+		Name:    snapshotName,
+		Created: time.Now().UTC().Format(time.RFC3339),
+		ID:      strconv.FormatInt(time.Now().UnixNano(), 10),
+	}
+	d.Snapshots[snapshotName] = snapshot
+
+	log.WithFields(log.Fields{
+		"backend":       d.Config.InstanceName,
+		"snapshotName":  snapshotName,
+		"sourceVolume":  internalVolName,
+		"requestedPool": vol.RequestedPool,
```

Comment:
The pool log lines here aren't especially valuable when creating a snapshot, I'd remove them.